src/coreclr/hosts/coreconsole/coreconsole.cpp

```diff
@@ -596,14 +596,25 @@ void showHelp() {
 int __cdecl wmain(const int argc, const wchar_t* argv[])
 {
 	auto programPath = _wcsdup(argv[0]);
-	auto extension = wcsrchr(programPath, '.') + 1;
-	if (wcscmp(extension, L"exe") != 0) {
-		::wprintf(W("This executable needs to have 'exe' extension"));
-		return -1;
+	auto extension = wcsrchr(programPath, '.');
+	if (extension == NULL) {
+		// We were called without the extension so need a bigger buffer
+		size_t pathLen = wcslen(programPath);
+		wchar_t *newPath = new wchar_t[pathLen + 5];
+		wcscpy_s(newPath, pathLen + 1, programPath);
```

Comment:
It would be more appropriate to specify the actual size of the buffer here. Maybe store pathLen + 5 to a local so that it is not repeated.